gammapy/scripts/image_pipe.py

```diff
@@ -175,13 +222,102 @@ def make_maps(self, radius, bkg_norm=True):
             Disk radius in pixels for the significance map.
         bkg_norm : bool
             If true, apply the scaling factor to the bkg map
+        spectral_index : float
+            Assumed power-law spectral index
+        exposure_weighted_spectrum : bool
+            True if you want that the total excess / exposure gives the integrated flux
         """
         self.make_total_counts()
         self.make_total_bkg(bkg_norm)
         self.make_significance(radius)
+        self.make_total_excess()
+        self.make_total_exposure(spectral_index, exposure_weighted_spectrum)
 
     def make_psf(self):
         log.info('Making PSF ...')
 
-    def make_exposure(self):
-        log.info('Making EXPOSURE ...')
+    def make_total_excess(self):
+        """Compute excess between counts and bkg map."""
+        self.maps["total_excess"] = SkyMap.empty_like(self.empty_image)
+        excess = self.maps["total_counts"].data - self.maps["total_bkg"].data
+        self.maps["total_excess"].data = excess
+
+    def make_exposure(self, obs_id, spectral_index=2.3, exposure_weighted_spectrum=False):
+        """Compute the exposure map for one observation.
+
+        Excess/exposure will give the differential flux at the energy Eref at the middle of the self.energy_band
+
+        If exposure_weighted_spectrum is true, it will give the integrated flux over the self.energy_band
+
+        Exposure is define as fallow:
+
+        .. math ::
+
+            EXPOSURE = \int A(E) \phi(E) * time_{obs} \, dE
+
+        assuming a power law for the flux :math:`\phi(E) = \phi_{Eref} \times \frac{E}{E_ref}^{spectral_index}`
```

Comment:
Let’s call it for_integral_flux and discuss later what is the best!!
Le 17 avr. 2016 à 22:29, Christoph Deil <notifications@github.com> a écrit :

> In gammapy/scripts/image_pipe.py:
> 
> > +        self.maps["total_excess"].data = excess
> > +
> > +    def make_exposure(self, obs_id, spectral_index=2.3, exposure_weighted_spectrum=False):
> > +        """Compute the exposure map for one observation.
> > +
> > +        Excess/exposure will give the differential flux at the energy Eref at the middle of the self.energy_band
> > +
> > +        If exposure_weighted_spectrum is true, it will give the integrated flux over the self.energy_band
> > +
> > +        Exposure is define as fallow:
> > +
> > +        .. math ::
> > +
> > +            EXPOSURE = \int A(E) \phi(E) * time_{obs} \, dE
> > +
> > +        assuming a power law for the flux :math:`\phi(E) = \phi_{Eref} \times \frac{E}{E_ref}^{spectral_index}`
> I currently get this:
> 
> 
> 
> I think it's better to choose symbols T and \Gamma in the formula and define them in the line after the formula.
> Also -- please add E_1 and E_2 to the integral, i.e. mention that the integration range is the energy band passed by the caller.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>